src/test/java/SalesforceModules/InvoiceUtil.java

```diff
@@ -0,0 +1,235 @@
+        boolean flag = false;
+        int size = 0;
+        KeywordUtil.delay(4000);
+        KeywordUtil.waitForVisible(InvoiceObject.selectstatus);
+        KeywordUtil.click(InvoiceObject.selectstatus,logStep);
+        KeywordUtil.delay(3000);
+        String xpath   ="(//lightning-base-combobox-item[@class='slds-media slds-listbox__option slds-media_center slds-media_small slds-listbox__option_plain'][contains(@data-value,'"+status+"')])[1]";
+        try {
+            size = KeywordUtil.getDriver().findElements(By.xpath(xpath)).size();
+
+        } catch (Exception e) {
+        }
+
+        if (size == 0) {
+            KeywordUtil.takeScreenshotAndAttachInReport();
+            System.out.println("no such status present");
+            System.out.println(size);
+        } else {
+            KeywordUtil.click(By.xpath(xpath), "status changed as Posted");
+        }
+
+    }
+
+    /**
+     *Generate invoices in billing module
+     *
+     *
+     * @param logStep the log
+     */
+    public static void generateInvoiceBilling(String taxStatus,String postedStatus, String logStep) throws InterruptedException {
+        KeywordUtil.clickJS_component(InvoiceObject.detailsTab,"Open Order");
+        KeywordUtil.clickJS_component(InvoiceObject.billNowEditButton,"Open Order");
+        KeywordUtil.clickJS_component(InvoiceObject.billNowButton,"Open Order");
+        QuoteUtil.clickSaveButton("clicked save button");
+        KeywordUtil.clickJS_component(InvoiceObject.relatedTab,"Open Order");
+        KeywordUtil.clickJS_component(InvoiceObject.invoices,"Click on Invoice");
+        KeywordUtil.clickJS_component(InvoiceObject.openInvoices,"Open the Invoice");
+
+        //code for tax info boxes
+        KeywordUtil.delay(3000);
+        KeywordUtil.clickJS_component(InvoiceObject.checkboxForTaxInfo,"edit button on tax info");
+        KeywordUtil.delay(3000);
+        setTaxInfo(taxStatus,"select tax status as None");
+        KeywordUtil.delay(3000);
+        KeywordUtil.clearInput(InvoiceObject.taxMessage);
+
+        //select status as posted
+        KeywordUtil.delay(2000);
+        setPostedStatus(postedStatus,"select status as Posted");
+        QuoteUtil.clickSaveButton("clicked save button");
+    }
+
+
+    /**
+     * Select the Account field for new payments
+     *
+     * @param logStep the log
+     * @param account the account name
+     */
+    public static void selectAccountPayments(String account, String logStep) throws InterruptedException {
+        boolean flag = false;
+        KeywordUtil.waitForVisible(ServiceSupportObject.searchAccounts_quote_ss);
+        KeywordUtil.click(ServiceSupportObject.searchAccounts_quote_ss, logStep);
+        KeywordUtil.inputText(ServiceSupportObject.searchAccounts_quote_ss, "Bill", logStep);
+        KeywordUtil.delay(5000);
+        Thread.sleep(5000);
+        String accname = "//lightning-base-combobox-formatted-text[contains(@title,'Bill')]";
```

Comment:
do we need this xpath here as it is not dynamic?

Reply:
done